src/main/java/com/minecolonies/coremod/colony/buildings/views/AbstractBuildingView.java

```diff
@@ -203,19 +203,19 @@ private void loadRequestSystemFromNBT(NBTTagCompound compound)
         this.citizensByRequests.keySet().forEach(citizen -> this.citizensByRequests.get(citizen).forEach(requestToken -> this.requestsByCitizen.put(requestToken, citizen)));
     }
 
-    @SuppressWarnings(Suppression.GENERIC_WILDCARD)
     public <R> ImmutableList<IRequest<? extends R>> getOpenRequestsOfType(@NotNull final CitizenDataView citizenData, final Class<R> requestType)
     {
         return ImmutableList.copyOf(getOpenRequests(citizenData).stream()
                                       .filter(request -> {
                                           Set<TypeToken> requestTypes = ReflectionUtils.getSuperClasses(request.getRequestType());
+                                          //TODO: Check on the types of this, it is TypeToken and Class, possible mismatch
                                           return requestTypes.contains(requestType);
                                       })
                                       .map(request -> (IRequest<? extends R>) request)
                                       .iterator());
     }
 
-    public ImmutableList<IRequest> getOpenRequests(@NotNull final CitizenDataView data)
+    public ImmutableList<IRequest<?>> getOpenRequests(@NotNull final CitizenDataView data)
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Remove usage of generic wildcard type. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1452)